source/dol/bases/d_CourseSelectGuide.cpp

```diff
@@ -0,0 +1,1000 @@
+    static const int textBoxIdxs[] = { T_lifeNumber_00, T_lifeNumber_01, T_lifeNumber_03, T_lifeNumber_02 };
+
+    for (int i = 0; i < PLAYER_COUNT; i++) {
+        daPyMng_c::PyType playerType = daPyCom_c::sc_PLAYER_ORDER[i];
+        int playerIndex = daPyMng_c::getPlayerIndex(playerType);
+        int rest = daPyMng_c::getRest(playerType);
+        if (mRest[i] != rest) {
+            mRest[i] = rest;
+            int textIdx = textBoxIdxs[i];
+            LytTextBox_c *textBox = mpTextBoxes[textIdx];
+            dGameCom::LayoutDispNumber(rest, dGameDisplay_c::c_PLAYNUM_DIGIT, textBox, true);
+        }
+    }
+    mShowRestNumber = false;
+}
+
+void dCourseSelectGuide_c::RestAlphaDisp() {
+    if (mRestAlpha == mRestAlphaTarget) {
+        return;
+    }
+    if (mRestAlphaTarget != 0) {
+        mRestAlpha += c_DISP_WAIT_TIMER;
+        if (mRestAlpha >= 255) {
+            mRestAlpha = 255;
+        }
+    } else {
+        mRestAlpha -= c_DISP_WAIT_TIMER;
+        if (mRestAlpha < 0) {
+            mRestAlpha = 0;
+        }
+    }
+
+    u8 alpha = mRestAlpha;
+    mpNullPanes[N_IconPos1P_00]->setAlpha(alpha);
+    mpPicturePanes[P_marioFace_00]->setAlpha(alpha);
+    mpNullPanes[N_IconPos2P_00]->setAlpha(alpha);
+    mpPicturePanes[P_luigiFace_00]->setAlpha(alpha);
+    mpNullPanes[N_IconPos3P_00]->setAlpha(alpha);
+    mpPicturePanes[P_BkinoFace_00]->setAlpha(alpha);
+    mpNullPanes[N_IconPos4P_00]->setAlpha(alpha);
+    mpPicturePanes[P_YkinoFace_00]->setAlpha(alpha);
+}
+
+void dCourseSelectGuide_c::execute() {
+    if (mInitialized && mEnabled) {
+        if (mShowRestNumber) {
+            RestNumberDisp();
+        }
+
+        mStateMgrWorldCourse.executeState();
+        mStateMgrGuide.executeState();
+
+        if (mMapView) {
+            mStateMgrUp.executeState();
+            mStateMgrDown.executeState();
+            mStateMgrLeft.executeState();
+            mStateMgrRight.executeState();
+
+            mMoveDirection = 0;
+            if (mEnableAllArrows) {
+                mEnableAllArrows = false;
+            }
+            if (mDisableArrows) {
+                if (
+                    !mLayout.isAnime(ANIM_OUT_ARROW_U) || !mLayout.isAnime(ANIM_OUT_ARROW_D) ||
+                    !mLayout.isAnime(ANIM_OUT_ARROW_L) || !mLayout.isAnime(ANIM_OUT_ARROW_R)
+                ) {
+                    mDisableArrows = false;
+                    mMapView = false;
+                }
+            }
+        }
+
+        mStateMgrScrollGuide.executeState();
+        mStateMgrShadow.executeState();
+        mLayout.AnimePlay();
+        ControllerConnectCheck();
+        RestAlphaDisp();
+        mLayout.calc();
+    }
+}
+
+void dCourseSelectGuide_c::draw() {
+    if (mInitialized && mEnabled) {
+        mLayout.entry();
+    }
+}
+
+void dCourseSelectGuide_c::doDelete() {
+    mLayout.doDelete();
+}
+
+void dCourseSelectGuide_c::CourseSelectSet(dWmLib::CourseType_e type) {
+    static const int sc_startPointIcons[] = {
```

Comment:
```suggestion
    static const int sc_startPointIcons[NORMAL_WORLD_COUNT + SPECIAL_WORLD_COUNT] = {
```
We can't use WORLD_COUNT because it's 10 unfortunately, but maybe we can define a new enum entry WORLD_USED_COUNT with this value.